plugins/clipboard/plugin.js

```diff
@@ -201,6 +203,37 @@
 				return dataTransfer.isFileTransfer() && dataTransfer.getFilesCount() === 1;
 			}
 
+			// Checking if the 'paste' event contains an image with one of the supported type on IE10+ (#4681)
+			function checkIEPasteImage( evt, dataTransfer, supportedImageTypes ) {
+				if ( evt.data.isIEFailed || !CKEDITOR.plugins.clipboard.isFileApiSupported || !CKEDITOR.env.ie ) {
+					return false;
+				}
+
+				var file = dataTransfer.getFile( 0 ) || false,
+					includeFileType = dataTransfer.getTypes() ? dataTransfer.getTypes()[ 0 ] : false,
+					eventFileType = file ? file.type : false,
+					transferType = dataTransfer.getTransferType( editor );
+
+				if ( eventFileType && includeFileType === 'Files' && transferType === CKEDITOR.DATA_TRANSFER_EXTERNAL ) {
+					return checkIfTypeIsSupported( supportedImageTypes, eventFileType );
+				} else {
+					return false;
+				}
+			}
+
+			// Add an attribute when the image is not loaded correctly. (#4681)
+			function setCustomIEEventAttribute( evt ) {
+				if ( CKEDITOR.env.ie && CKEDITOR.plugins.clipboard.isFileApiSupported ) {
```

Comment:
The second check is redundant because this code will run only in environments that support File API.